File: Calorimeter/CaloClusterCorrection/python/CaloSwEtamod_g3.py
# Copyright (C) 2002-2017 CERN for the benefit of the ATLAS collaboration

#
# $Id: CaloSwEtamod_g3.py,v 1.2 2006-11-16 07:07:45 ssnyder Exp $
#
# File: CaloClusterCorrection/python/CaloSwEtamod_g3.py
# Created: Nov 2006, sss
# Purpose: Eta modulation corrections, original G3 version.
#
# This is the original G3-based eta modulation correction, translated from the
# original fortran version.
#


class CaloSwEtamod_g3_parms:
    eta_size = 3
    correction = [
        [ [1.0052, -1.39, 56.0],     # barrel
          [1.0005, -0.19,  7.0],
          [1.0001, -0.137, 6.4] ],

        [ [1.011,  -2.2,  75.0],     # endcap
          [1.0045, -0.7,  22.0],
          [1.0027, -0.37,  9.0] ]
        ]

File: Calorimeter/CaloClusterCorrection/src/CaloSwEtamod_g3.cxx
/*
  Copyright (C) 2002-2020 CERN for the benefit of the ATLAS collaboration
*/

/********************************************************************

NAME:     CaloSwEtamod_g3.cxx
PACKAGE:  offline/Calorimeter/CaloClusterCorrection

AUTHORS:  H. Ma, S. Rajagopalan
CREATED:  Dec. 15, 1999

PURPOSE:  correction for the modulations with the position in eta
          (Tuned using 50 GeV Et photons)  
          base class: CaloClusterCorrection
          Correction tuned on G3 samples.

Atrecon Orig: emreco/qetamod.F

Updated:  May 10, 2000    (SR, HM)
          Migrated to Athena Framework from PASO

Updated:  Jan 5, 2001    (HM)
          QA. 

Updated:  May 5, 2004    (Sven Menke)
	  base class changed from algo to tool

Updated:  June, 2004    (sss)
          Use ToolWithConstants to get correction constants.

********************************************************************/
// include header files 
#include "CaloSwEtamod_g3.h"


using xAOD::CaloCluster;


// make correction to one cluster 
void CaloSwEtamod_g3::makeCorrection (const Context& myctx,
                                      CaloCluster* cluster) const
{
  float aeta = fabs (cluster->etaBE(2));
  if (aeta > 2.5) return;

  int ieta = m_neta(myctx)/2;
  if (ieta > 3) return;       // something went wrong

  CaloRec::Array<1> coef = m_correction(myctx)[aeta < 1.5 ? 0 : 1][ieta-1];

  float u = fmod (aeta, 0.025); 
  float qetamod = coef[0] + coef[1]*u + coef[2]*u*u;

  setenergy (cluster, qetamod * cluster->e());
}

